perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1213 @@
+import simtk.openmm as openmm
+import simtk.openmm.app as app
+import simtk.unit as unit
+import mdtraj as md
+import numpy as np
+import copy
+import enum
+
+ONE_4PI_EPS0 = 138.935456
+InteractionGroup = enum.Enum("InteractionGroup", ['unique_old', 'unique_new', 'core', 'environment'])
+
+class HybridTopologyFactory(object):
+    """
+    This class generates a hybrid topology based on a perses topology proposal. This class treats atoms
+    in the resulting hybrid system as being from one of four classes:
+
+    unique_old_atom : these atoms are not mapped and only present in the old system. Their interactions will be on for
+        lambda=0, off for lambda=1
+    unique_new_atom : these atoms are not mapped and only present in the new system. Their interactions will be off
+        for lambda=0, on for lambda=1
+    core_atom : these atoms are mapped, and are part of a residue that is changing. Their interactions will be those
+        corresponding to the old system at lambda=0, and those corresponding to the new system at lambda=1
+    environment_atom : these atoms are mapped, and are not part of a changing residue. Their interactions are always
+        on and are alchemically unmodified.
+    """
+
+    def __init__(self, topology_proposal, current_positions, new_positions):
+        """
+        Initialize the Hybrid topology factory.
+
+        Parameters
+        ----------
+        topology_proposal : perses.rjmc.topology_proposal.TopologyProposal object
+            TopologyProposal object rendered by the ProposalEngine
+        current_positions : [n,3] np.ndarray of float
```

Comment:
Do these *have* to be stripped of units, or are unit-bearing quantities allowed too?
Same for new_positions.